client/graphviz-viewer.js

```diff
@@ -22,8 +22,18 @@ class GraphvizViewer extends HTMLElement {
     this.attachShadow({mode: 'open'})
     this.shadowRoot.appendChild(template.content.cloneNode(true));
     this.message('drawing diagram');
+    this._svg = '';
   }
 
+  get dot() {
+    // see https://stackoverflow.com/a/34064434/1074208
+    return new DOMParser()
+      .parseFromString(this.innerHTML, 'text/html')
+      .documentElement
+      .textContent;
+  }
```

Comment:
On reflection, this feels backwards. Think we should prefer to save the dot before it has been inserted into the DOM. That would mean computing the value in the DOM from an internal property with the original dot.

Reply:
Looking more closely at the code in context, and also at the commit history... for basic usage of this custom element, the DOM is really the source of truth.

Here's an example which reveals the trade-off. This is valid HTML with the `>` characters written with HTML encoding.

    <graphviz-viewer>
      strict digraph { A -&gt; {B C} -&gt; D }
    </graphviz-viewer>

The dot we want to hand to graphviz needs to decode the characters into ASCII:

    strict digraph { A -> {B C} -> D }

As written, `get dot()` does The Right Thing™ for hand-coded uses of the custom element.